index.js

```diff
@@ -1,43 +1,56 @@
+import { getIndexProviderPeerIdFromSmartContract } from './lib/smart-contract-client.js'
+import { getIndexProviderPeerIdFromFilecoinMinerInfo } from './lib/filecoin-rpc-client.js'
+import { rpc } from './lib/rpc-client.js'
 export { MINER_TO_PEERID_CONTRACT_ADDRESS, MINER_TO_PEERID_CONTRACT_ABI } from './lib/constants.js'
+
 /**
  * @param {string} minerId A miner actor id, e.g. `f0142637`
+ * @param {import('ethers').Contract} smartContract The smart contract instance
  * @param {object} options
  * @param {number} [options.maxAttempts]
- * @returns {Promise<string>} Miner's PeerId, e.g. `12D3KooWMsPmAA65yHAHgbxgh7CPkEctJHZMeM3rAvoW8CZKxtpG`
+ * @param {string} [options.rpcUrl]
+ * @param {string} [options.rpcAuth]
+ * @param {(method:string,params:unknown[])=> Promise<unknown> } [options.rpcFn] The RPC function to use
+ * @returns {Promise<{ peerId: string, source: 'smartContract'|'filecoinMinerInfo' }>} Miner's PeerId, e.g. `12D3KooWMsPmAA65yHAHgbxgh7CPkEctJHZMeM3rAvoW8CZKxtpG` and the source of the data
```

Comment:
I find the "filecoin" prefix redundant, this library does not deal with other blockchains than Filecoin.

```suggestion
 * @returns {Promise<{ peerId: string, source: 'smartContract'|'minerInfo' }>} Miner's PeerId, e.g. `12D3KooWMsPmAA65yHAHgbxgh7CPkEctJHZMeM3rAvoW8CZKxtpG` and the source of the data
```

Reply:
+1 to removing Filecoin here.

Reply:
Addressed in f693428